tests/autobahn/client.py

```diff
@@ -6,31 +6,32 @@
 
 
 async def client(loop, url, name):
-    ws = await aiohttp.ws_connect(url + "/getCaseCount")
-    num_tests = int((await ws.receive()).data)
-    print("running %d cases" % num_tests)
-    await ws.close()
-
-    for i in range(1, num_tests + 1):
-        print("running test case:", i)
-        text_url = url + "/runCase?case=%d&agent=%s" % (i, name)
-        ws = await aiohttp.ws_connect(text_url)
-        while True:
-            msg = await ws.receive()
-
-            if msg.type == aiohttp.WSMsgType.TEXT:
-                await ws.send_str(msg.data)
-            elif msg.type == aiohttp.WSMsgType.BINARY:
-                await ws.send_bytes(msg.data)
-            elif msg.type == aiohttp.WSMsgType.CLOSE:
-                await ws.close()
-                break
-            else:
-                break
-
-    url = url + "/updateReports?agent=%s" % name
-    ws = await aiohttp.ws_connect(url)
-    await ws.close()
+    async with aiohttp.ClientSession() as session:
+        async with session.ws_connect(url + "/getCaseCount") as ws:
+            num_tests = int((await ws.receive()).data)
+            print("running %d cases" % num_tests)
+            await ws.close()
```

Comment:
`ws` object automatically closes after the exit from the `session.ws_connect` context manager. So maybe it makes sense to replace all of the `ws_connect / ws.close()` pairs with separate context managers, like so:

```python3
async with session.ws_connect(url + "/getCaseCount") as ws:
    num_tests = int((await ws.receive()).data)
    print("running %d cases" % num_tests)
    
for i in range(1, num_tests + 1):
    ...
    async with session.ws_connect(text_url) as ws:
        ...
 ```

Reply:
Not sure why I didn't notice the assignment to `ws` nested in the outer `ws` context, or why the testclient returned OK on all the test cases, but I've made these changes :)